src/omv/modules/py_tf.c

```diff
@@ -635,6 +831,10 @@ STATIC mp_obj_t py_tf_detect(uint n_args, const mp_obj_t *pos_args, mp_map_t *kw
     rectangle_t roi = py_helper_arg_to_roi(args[ARG_roi].u_obj, image);
     bool invert = args[ARG_invert].u_int;
 
+    if ((args[ARG_model].u_int < 0) || (args[ARG_model].u_int >= PY_TF_MODEL_COUNT)) {
+        mp_raise_msg(&mp_type_ValueError, MP_ERROR_TEXT("Invalid model type!"));
+    }
+
```

Comment:
This should now set the model type based on number of ops.
```suggestion
    if ((args[ARG_model].u_int == -1)) {
        switch (model->params->operators_size) {
           case PY_TF_MODEL_FOMO:
               model_type = PY_TF_MODEL_FOMO;
               break;
            default:
               mp_raise_msg(&mp_type_ValueError, MP_ERROR_TEXT("Please specify the model type using type=... "));
        }
    }
```

Reply:
Done, just need to collect signatures.